tests/support/server.rs

```diff
@@ -345,11 +345,14 @@ where
     .unwrap()
 }
 
-async fn low_level_server_client<F>(mut client_socket: TcpStream, do_response: F)
-where
+async fn low_level_server_client<F>(
+    mut client_socket: TcpStream,
+    requests_count: usize,
+    do_response: F,
+) where
     for<'c> F: Fn(&'c [u8], &'c mut TcpStream) -> Box<dyn Future<Output = ()> + Send + 'c>,
 {
-    loop {
+    for _ in 0..requests_count {
```

Comment:
I'm not sure about this change. Previously, this loop never exited until the test server was dropped.

Reply:
Hm, does the behavior from tower-http require that this loop exit earlier?

Reply:
I think the behavior in `master` is the correct one. If there are extra bytes in chunked encoding, it should error when decompression is enabled and hang when decompression is disabled.

At least, that is how curl behaves

```
$ curl --compressed --output - 127.0.0.1:8000
curl: (23) Failed reading the chunked-encoded stream
hello world from malformed chunked gzip test

$ curl --output - 127.0.0.1:8000
��H���W(�/�IQH+��U�M�I�/�MMQH�(����U�
%��%�&L�,0

 # hangs
```

reqwest always had this behaviour (checked eb94f26919e881177ad3d9cd172f47d1a8263799) so I'm worried that merging this now would be a disruptive change.